src/lib.rs

```diff
@@ -3370,7 +3353,7 @@ impl Build {
         // CROSS_COMPILE is of the form: "arm-linux-gnueabi-"
         let cc_env = self.getenv("CROSS_COMPILE");
         let cross_compile = cc_env.as_ref().map(|s| s.trim_end_matches('-').to_owned());
-        cross_compile.or(linker_prefix).or(match &target[..] {
+        cross_compile.or(linker_prefix).or(match target {
```

Comment:
IMO we'd need

```suggestion
        cross_compile.or(linker_prefix).or_else(|| match target {
```

Reply:
Done. Changing it does affect the formatting.